### PR TITLE
Remove legacy taxons

### PR DESCRIPTION
**Pull request by hannako**
[⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️]

## What

Removes all code referring to legacy taxons and the ability to add any to content items.

## Why

We are retiring Specialist Topics (which is what we refer to legacy taxons as outside of code) and don't want to add to the pile that we need to remove. 

[Trello](https://trello.com/c/7fSxKIyS/1257-remove-the-ui-for-setting-a-legacy-taxon-from-content-tagger-s)

## Before

content-tagger integration publishing service gov uk_taxons_new

## After

content-tagger dev gov uk_taxons_new

### CODE REVIEW

**Comment by KludgeKML** [comment]
Do you still need this line? 
https://github.com/alphagov/content-tagger/blob/remove_legacy_taxons/spec/features/taxonomy_editing_spec.rb#L327

**Review by KludgeKML — APPROVED** [review]
LGTM, could maybe remove the line in my comment?

**Review by ollietreend — COMMENTED** [review]
Nice work 🙌🏻 thanks for doing this work.

This makes me especially happy: 3 lines added, 112 lines removed

My domain knowledge of taxonomies and this app in general is quite limited, so no need to act upon my comments if you don't think they're helpful!

File: app/services/taxonomy/update_taxon.rb [hunk 1]
```diff
@@ -57,14 +56,6 @@ def payload(locale = "en")
       Taxonomy::BuildTaxonPayload.call(taxon: taxon, locale: locale)
     end
 
-    def legacy_taxon_ids
-      return [] if taxon.legacy_taxons.blank?
-
-      Array(
-        Tagging::BasePathLookup.find_by_base_paths(taxon.legacy_taxons),
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply app/services/taxonomy/update_taxon.rb:64]: Yeah it's used for looking up Related Links it looks like. I'm leaving it in.

**Pull request merged**